extensions/banner/runtime/src/main/java/io/quarkus/banner/runtime/BannerRecorder.java

```diff
@@ -0,0 +1,43 @@
+package io.quarkus.banner.runtime;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.builder.Version;
+import io.quarkus.deployment.util.FileUtil;
+import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.runtime.annotations.Recorder;
+
+@Recorder
+public class BannerRecorder {
+    private Logger logger = Logger.getLogger(BannerRecorder.class);
+
+    public RuntimeValue<Optional<Supplier<String>>> provideBannerSupplier(BannerConfig config) {
+        Supplier<String> supplierValue = () -> readBannerFile(config);
+        return new RuntimeValue<>(Optional.of(supplierValue));
+    }
+
+    private String readBannerFile(BannerConfig config) {
```

Comment:
Can this be done during build time? And pass the banner text in the recorder? 
i.e modify the signature of `provideBannerSupplier` to  
```
 public RuntimeValue<Optional<Supplier<String>>> provideBannerSupplier(String bannerText) {
        Supplier<String> supplierValue = () -> bannerText;
        return new RuntimeValue<>(Optional.of(supplierValue));
    }
```

Reply:
Actually david said this 
```
This supplier should do the actual banner file reading, so that the banner is only held in memory when it is actually being displayed.
```

Reply:
But for native image part, It makes sense, we do not have to use nativeImageBuildItem then

Reply:
umm So how to proceed now ?

Reply:
I think we can probably just leave it as it is for now. This isn't terribly important IMHO

Reply:
Lemme make the final changes as for now. Also new suggested changes makes it easier to read the code more. :)

Reply:
Oh yes, let's keep it this way for now.